### PR TITLE
Use suggested kernel if we have only one kernel

### LINKED ISSUE #136547: Never prompted to select a kernel in Notebooks
When creating new notebooks we're never prompted to select a kernel.

```typescript
	getSelectedOrSuggestedKernel(notebook: INotebookTextModel): INotebookKernel | undefined {
		// returns SELECTED or the ONLY available kernel
		const info = this._notebookKernelService.getMatchingKernel(notebook);
		return info.selected ?? info.suggestions[0];
	}
```

Should be 
```typescript
	getSelectedOrSuggestedKernel(notebook: INotebookTextModel): INotebookKernel | undefined {
		// returns SELECTED or the ONLY available kernel
		const info = this._notebookKernelService.getMatchingKernel(notebook);
		return info.selected ?? (info.suggestions.length === 1 : info.suggestions[0] : undefined);
	}
```

### PR DESCRIPTION
**Pull request by DonJayamanne**
This PR fixes #136547

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class NotebookEditorKernelManager extends Disposable {
 	getSelectedOrSuggestedKernel(notebook: INotebookTextModel): INotebookKernel | undefined {
 		// returns SELECTED or the ONLY available kernel
```
Comment (IanMatthewHuff) [inline src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts:26]: Minor, but this comment looks like it should be updated along with this change. It returns selected or the suggested kernel if there is only one suggestion.

**Review by IanMatthewHuff — APPROVED** [review]
Looks good to me with one comment update.

File: src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts:26]: Oh yes, that's still a bug then, thantks

Comment (IanMatthewHuff) [reply src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts:26]: I'm not sure if that update is correct. I was thinking that your code was right before but the comment in the function was wrong. Is that not the case? I was reading this fix as:

Selected if available. If not, suggested if only one suggested.

Which is what you had before. What it's changed to now you would return undefined if there was only one suggested and 2+ total kernels.

Comment (DonJayamanne) [reply src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts:26]: Agreed, will do that.
Looks like I'll need to submit this from another machine (my wsl where my commits are signed & SSH setup properly)

**Comment by DonJayamanne** [comment]
Looks like I'll need to submit this from another machine (my wsl where my commits are signed & SSH setup properly)

**Review by IanMatthewHuff — APPROVED** [review]

File: src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/vs/workbench/contrib/notebook/browser/notebookEditorKernelManager.ts:26]: We cannot, 
* If user has only one kernel, we should use that withouth prompting
* If user has a selected kernel, we should use that without prompting
* If user doesn't have any kiernels, do nothing
* In all other cases we should prompt the user to select a kernel

**Review by raghavthind2005 — APPROVED** [review]
the code looks good to me too!

**Pull request merged**